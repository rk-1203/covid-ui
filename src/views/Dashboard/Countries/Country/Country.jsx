import { COUNTRY_SUMMARY_TABLE_COLUMNS } from "../../../../constants/constant";
import "./Country.scss";

function Country(props) {

    return (
        <tr className="country-row">
            {
                COUNTRY_SUMMARY_TABLE_COLUMNS.map(({id, displayName, formatter = v => v}) => (
                    <td data-th={displayName} key={id}>{formatter(props[id])}</td>
                ))
            }
        </tr>
    )

}

export default Country;
