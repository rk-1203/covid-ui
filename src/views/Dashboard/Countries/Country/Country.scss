.country-row {
    border-top: 1px solid #ddd;
    border-bottom: 1px solid #ddd;
    background-color: #f5f9fc;
    border-color: #bfbfbf;
    &:nth-child(odd) {
        background-color: #ebf3f9;
    }
    &:first-child {
        @media screen and (max-width: 601px) {
            border-top: none;
        }
    }
    @media screen and (min-width: 600px) {
        display: grid;
        grid-template-columns: 1.5fr repeat(6, 1fr);
        &:hover {
            background-color: #d8e7f3;
        }
    }
    td {
        display: block;
        text-align: left;
        padding: 10px 20px;
        &:first-child {
            margin-top: 4px;
            @media screen and (min-width: 600px) {
                // padding-left: 0;
            }
        }
        &:last-child {
            margin-bottom: 4px;
            @media screen and (min-width: 600px) {
                // padding-right: 0;
            }
        }
        &:before {
            content: attr(data-th) ": ";
            font-weight: bold;
            width: 160px;
            display: inline-block;
            color: #000;
        }
        @media screen and (min-width: 600px) {
            display: table-cell;
            padding: 20px 0;
            text-align: center;
            &:before {
                display: none;
            }
        }
    }
}