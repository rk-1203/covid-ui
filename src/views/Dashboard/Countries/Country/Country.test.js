import React from 'react';
import { render } from '@testing-library/react';

import Country from "./Country";

const props = {
    ID: "e0296b5d-0c21-484d-be0f-7ee48a5f9a7b",
    Country: "Afghanistan",
    CountryCode: "AF",
    Slug: "afghanistan",
    NewConfirmed: 0,
    TotalConfirmed: 167739,
    NewDeaths: 0,
    TotalDeaths: 7454,
    NewRecovered: 0,
    TotalRecovered: 0,
    Date: "2022-02-09T16:49:33.508Z",
    Premium: {}
}

describe("Country", () => {
    it("renders properly", () => {
        const { container } = render(
            <table>
                <tbody>
                    <Country {...props} />
                </tbody>
            </table>
        );
        expect(container.firstChild).toMatchSnapshot();
    })
})