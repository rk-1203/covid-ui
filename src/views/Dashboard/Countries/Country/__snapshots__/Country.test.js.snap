// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country renders properly 1`] = `
<table>
  <tbody>
    <tr
      class="country-row"
    >
      <td
        data-th="Country"
      >
        Afghanistan
      </td>
      <td
        data-th="New Confirmed"
      >
        0
      </td>
      <td
        data-th="Total Confirmed"
      >
        1,67,739
      </td>
      <td
        data-th="New Deaths"
      >
        0
      </td>
      <td
        data-th="Total Deaths"
      >
        7,454
      </td>
      <td
        data-th="New Recovered"
      >
        0
      </td>
      <td
        data-th="Total Recovered"
      >
        0
      </td>
    </tr>
  </tbody>
</table>
`;
